packages/fe-tw/src/hooks/useBuildingAudit.ts

```diff
@@ -0,0 +1,117 @@
+"use client";
+
+import type { AuditData } from "@/consts/audit";
+import {
+   getAuditRecordDetails,
+   getAuditRecordIdsForBuilding,
+} from "@/services/auditRegistryService";
+import { fetchJsonFromIpfs } from "@/services/ipfsService";
+import { useMutation, useQuery } from "@tanstack/react-query";
+import useWriteContract from "@/hooks/useWriteContract";
+import { useEffect } from "react";
+import { useExecuteTransaction } from "./useExecuteTransaction";
+import { auditRegistryAbi } from "@/services/contracts/abi/auditRegistryAbi";
+import { ContractId } from "@hashgraph/sdk";
+import { AUDIT_REGISTRY_ADDRESS } from "@/services/contracts/addresses";
+
+export function useBuildingAudit(buildingAddress: `0x${string}`) {
+   const { executeTransaction } = useExecuteTransaction();
+   const { writeContract } = useWriteContract();
+   
+   const { data: auditData, isLoading: auditDataLoading } = useQuery<{ data: AuditData, recordId: bigint } | null>({
+      queryKey: ["auditData", `auditData_${buildingAddress}`],
+      queryFn: async () => {
+         const data = await loadAuditData();
+
+         return data;
+      },
+      enabled: !!buildingAddress,
+   });
+   
+   async function loadAuditData() {
```

Comment:
No need for separate function of `loadAuditData` just out content into the `auditData` useQuery